### ISSUE TITLE
FileNotFoundException

### ISSUE THREAD

**Issue by mibby** [Gnat008/PerWorldInventory#191]
PWI compiled as of this commit. https://github.com/Gnat008/PerWorldInventory/commit/67f260e47f97c15c38622c58450aa7d63625f5d9

I deleted all /data/ folder files to reset inventories for testing. This error occurred at start-up. Restarting the server does not resolve the problem.

```
[09:40:56] [Server thread/INFO]: [PerWorldInventory] Enabling PerWorldInventory v1.9.2
[09:40:56] [Server thread/WARN]: [PerWorldInventory] Could not save default.json to plugins/PerWorldInventory/default.json because default.json already exists.
[09:40:56] [Server thread/INFO]: [PerWorldInventory] Registering listeners...
[09:40:56] [Server thread/INFO]: [PerWorldInventory] Listeners registered!
[09:40:56] [Server thread/ERROR]: [PerWorldInventory] An error occurred copying file '__default.json' to 'Build.json': [FileNotFoundException] plugins/PerWorldInventory/data/defaults/__default.json (No such file or directory)
[09:40:56] [Server thread/WARN]: java.io.FileNotFoundException: plugins/PerWorldInventory/data/defaults/__default.json (No such file or directory)
[09:40:56] [Server thread/WARN]: 	at java.io.FileInputStream.open0(Native Method)
[09:40:56] [Server thread/WARN]: 	at java.io.FileInputStream.open(FileInputStream.java:195)
[09:40:56] [Server thread/WARN]: 	at java.io.FileInputStream.<init>(FileInputStream.java:138)
[09:40:56] [Server thread/WARN]: 	at me.gnat008.perworldinventory.util.FileUtils.copyFile(FileUtils.java:18)
[09:40:56] [Server thread/WARN]: 	at me.gnat008.perworldinventory.groups.GroupManager.setDefaultsFile(GroupManager.java:187)
[09:40:56] [Server thread/WARN]: 	at me.gnat008.perworldinventory.groups.GroupManager.loadGroupsToMemory(GroupManager.java:155)
[09:40:56] [Server thread/WARN]: 	at me.gnat008.perworldinventory.PerWorldInventory.onEnable(PerWorldInventory.java:128)
[09:40:56] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:271)
[09:40:56] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:316)
[09:40:56] [Server thread/WARN]: 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:407)
[09:40:56] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_11_R1.CraftServer.enablePlugin(CraftServer.java:378)
[09:40:56] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_11_R1.CraftServer.enablePlugins(CraftServer.java:328)
[09:40:56] [Server thread/WARN]: 	at net.minecraft.server.v1_11_R1.MinecraftServer.t(MinecraftServer.java:437)
[09:40:56] [Server thread/WARN]: 	at net.minecraft.server.v1_11_R1.MinecraftServer.l(MinecraftServer.java:398)
[09:40:56] [Server thread/WARN]: 	at net.minecraft.server.v1_11_R1.MinecraftServer.a(MinecraftServer.java:336)
[09:40:56] [Server thread/WARN]: 	at net.minecraft.server.v1_11_R1.DedicatedServer.init(DedicatedServer.java:276)
[09:40:56] [Server thread/WARN]: 	at net.minecraft.server.v1_11_R1.MinecraftServer.run(MinecraftServer.java:603)
[09:40:56] [Server thread/WARN]: 	at java.lang.Thread.run(Thread.java:745)
[09:40:56] [Server thread/ERROR]: [PerWorldInventory] An error occurred copying file '__default.json' to 'Event.json': [FileNotFoundException] plugins/PerWorldInventory/data/defaults/__default.json (No such file or directory)
[09:40:56] [Server thread/WARN]: java.io.FileNotFoundException: plugins/PerWorldInventory/data/defaults/__default.json (No such file or directory)
[09:40:56] [Server thread/WARN]: 	at java.io.FileInputStream.open0(Native Method)
[09:40:56] [Server thread/WARN]: 	at java.io.FileInputStream.open(FileInputStream.java:195)
[09:40:56] [Server thread/WARN]: 	at java.io.FileInputStream.<init>(FileInputStream.java:138)
[09:40:56] [Server thread/WARN]: 	at me.gnat008.perworldinventory.util.FileUtils.copyFile(FileUtils.java:18)
[09:40:56] [Server thread/WARN]: 	at me.gnat008.perworldinventory.groups.GroupManager.setDefaultsFile(GroupManager.java:187)
[09:40:56] [Server thread/WARN]: 	at me.gnat008.perworldinventory.groups.GroupManager.loadGroupsToMemory(GroupManager.java:155)
[09:40:56] [Server thread/WARN]: 	at me.gnat008.perworldinventory.PerWorldInventory.onEnable(PerWorldInventory.java:128)
[09:40:56] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:271)
[09:40:56] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:316)
[09:40:56] [Server thread/WARN]: 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:407)
[09:40:56] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_11_R1.CraftServer.enablePlugin(CraftServer.java:378)
[09:40:56] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_11_R1.CraftServer.enablePlugins(CraftServer.java:328)
[09:40:56] [Server thread/WARN]: 	at net.minecraft.server.v1_11_R1.MinecraftServer.t(MinecraftServer.java:437)
[09:40:56] [Server thread/WARN]: 	at net.minecraft.server.v1_11_R1.MinecraftServer.l(MinecraftServer.java:398)
[09:40:56] [Server thread/WARN]: 	at net.minecraft.server.v1_11_R1.MinecraftServer.a(MinecraftServer.java:336)
[09:40:56] [Server thread/WARN]: 	at net.minecraft.server.v1_11_R1.DedicatedServer.init(DedicatedServer.java:276)
[09:40:56] [Server thread/WARN]: 	at net.minecraft.server.v1_11_R1.MinecraftServer.run(MinecraftServer.java:603)
[09:40:56] [Server thread/WARN]: 	at java.lang.Thread.run(Thread.java:745)
[09:40:56] [Server thread/INFO]: [PerWorldInventory] Registering commands...
[09:40:56] [Server thread/INFO]: [PerWorldInventory] Commands registered!
[09:40:56] [Server thread/INFO]: [PerWorldInventory] Vault found! Hooking into it...
[09:40:56] [Server thread/INFO]: [PerWorldInventory] Hooked into Vault!
```

Loading into a separate world which has its own inventory group causes the same exception.
```
[09:49:12] [Craft Scheduler Thread - 9/ERROR]: [PerWorldInventory] Unable to find inventory data for player 'player' for group 'Build': [FileNotFoundException] plugins/PerWorldInventory/data/defaults/__default.json (No such file or directory)
[09:49:12] [Craft Scheduler Thread - 9/WARN]: java.io.FileNotFoundException: plugins/PerWorldInventory/data/defaults/__default.json (No such file or directory)
[09:49:12] [Craft Scheduler Thread - 9/WARN]: 	at java.io.FileInputStream.open0(Native Method)
[09:49:12] [Craft Scheduler Thread - 9/WARN]: 	at java.io.FileInputStream.open(FileInputStream.java:195)
[09:49:12] [Craft Scheduler Thread - 9/WARN]: 	at java.io.FileInputStream.<init>(FileInputStream.java:138)
[09:49:12] [Craft Scheduler Thread - 9/WARN]: 	at java.io.FileReader.<init>(FileReader.java:72)
[09:49:12] [Craft Scheduler Thread - 9/WARN]: 	at me.gnat008.perworldinventory.data.FileWriter.getFromDefaults(FileWriter.java:179)
[09:49:12] [Craft Scheduler Thread - 9/WARN]: 	at me.gnat008.perworldinventory.data.FileWriter.lambda$getFromDatabase$2(FileWriter.java:138)
[09:49:12] [Craft Scheduler Thread - 9/WARN]: 	at org.bukkit.craftbukkit.v1_11_R1.scheduler.CraftTask.run(CraftTask.java:58)
[09:49:12] [Craft Scheduler Thread - 9/WARN]: 	at org.bukkit.craftbukkit.v1_11_R1.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:52)
[09:49:12] [Craft Scheduler Thread - 9/WARN]: 	at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22)
[09:49:12] [Craft Scheduler Thread - 9/WARN]: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[09:49:12] [Craft Scheduler Thread - 9/WARN]: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[09:49:12] [Craft Scheduler Thread - 9/WARN]: 	at java.lang.Thread.run(Thread.java:745)
```

Edit: I manually copied the Default.json file and renamed it to Build.json and Event.json (my separate inventory groups), then put it in the default data folder. Seems to start-up fine now with the exception of 

```
[09:54:14] [Server thread/WARN]: [PerWorldInventory] Could not save default.json to plugins/PerWorldInventory/default.json because default.json already exists.
```

but you may want to fix the problem of default json group files not generating.

**Comment by PinoyLifeCraft** [comment]
Same here !

> ERROR An error occurred copying file '__default.json' to 'hub.json': [FileNotFoundException] plugins/PerWorldInventory/data/defaults/__default.json (No such file or directory)

**Comment by PockyMonstrosity** [comment]
Same here. Renamed PerWorldInventory/data/defaults/default.json to PerWorldInventory/data/defaults/__default.json and it appears to create everything correctly. At least it no longer throws the exception.

**Comment by Gnat008** [comment]
Fixed in v1.11.
